pcl_config.h.in

```diff
@@ -22,13 +22,18 @@
 
 /* Index type and signed/unsigned property */
 #define PCL_INDEX_SIGNED ${PCL_INDEX_SIGNED_STR}
+#define PCL_INDEX_SIZE_TEMP ${PCL_INDEX_SIZE}
 
-#if PCL_MINOR_VERSION <= 11
-// sizeof returns bytes, while we measure size by bits in the template
-#define PCL_INDEX_SIZE (sizeof(int) * 8)
+#if (PCL_INDEX_SIZE_TEMP > 0)
+  #define PCL_INDEX_SIZE ${PCL_INDEX_SIZE}
```

Comment:
Using `PCL_INDEX_SIZE_TEMP` might be better instead of `${PCL_INDEX_SIZE}`

Reply:
It looks like I can't follow both of these suggestions together. If I undefine `PCL_INDEX_SIZE_TEMP`, then at compile time `PCL_INDEX_SIZE` cannot be set to it, because it is no longer available.

Reply:
Oh, yeah. Can we just remove the temporary variable in that case? It's sole use seems to be comparison with 0

Reply:
I'm unresolving the conversation, because `PCL_INDEX_SIZE_TEMP` still needs to be removed.

Reply:
My bad, I thought by remove @kunaltyagi meant I should undef `PCL_INDEX_SIZE_TEMP`, not get rid of it altogether.
 Is this what you had in mind?
```c++
#if (${PCL_INDEX_SIZE} > 0)
  #define PCL_INDEX_SIZE ${PCL_INDEX_SIZE}
#else
  #if PCL_MINOR_VERSION <= 11
    // sizeof returns bytes, while we measure size by bits in the template
    #define PCL_INDEX_SIZE (sizeof(int) * 8)
  #else
    #define PCL_INDEX_SIZE 32
  #endif //PCL_MINOR_VERSION
#endif 
```

Reply:
That's what I interpreted, because there's no need the actually define anything.